lints/lint_dnsname_contains_bare_iana_suffix.go

```diff
@@ -0,0 +1,43 @@
+// lint_dnsname_contains_bare_iana_suffix.go
+
+package lints
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+)
+
+type dnsNameContainsBareIANASuffix struct{}
+
+func (l *dnsNameContainsBareIANASuffix) Initialize() error {
+	return nil
+}
+
+func (l *dnsNameContainsBareIANASuffix) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubscriberCert(c) && util.DNSNamesExist(c)
+}
+
+func (l *dnsNameContainsBareIANASuffix) Execute(c *x509.Certificate) *LintResult {
+	if c.Subject.CommonName != "" {
+		if util.IsInTLDMap(c.Subject.CommonName) {
```

Comment:
Why do we use the map instead of the parse function in `util`

Reply:
I thought you wanted IANA suffix not ICANN Public Suffix? 

The `util.ICANNPublicSuffixParse` is just the public part of the public suffix

Reply:
OK, but still let @zakird do a pass